src/patterns/ec2-app.ts

```diff
@@ -393,11 +393,16 @@ export class GuEc2App {
     }
 
     if (!props.monitoringConfiguration.noMonitoring) {
-      new Gu5xxPercentageAlarm(scope, "Alarm", {
+      new Gu5xxPercentageAlarm(scope, {
         app,
         loadBalancer,
         ...props.monitoringConfiguration,
       });
+      new GuUnhealthyHostsAlarm(scope, {
```

Comment:
Since users don't need to supply any props to get this working (we only need an `snsTopicName` and the users has already provided one to get `Gu5xxPercentageAlarm` working), I have not changed the pattern's API at all. At the very least, I need to add docs for this, however there are a few other options:

1. Add a top-level boolean which allows users to opt-in/opt-out of this alert, regardless of whether they want a `Gu5xxPercentageAlarm` or not.
2. Change the type of `monitoringConfiguration: NoMonitoring | Gu5xxPercentageMonitoringProps;` to cover more alarms e.g. `monitoringConfiguration: NoMonitoring | GuSomeMoreGenericMonitoringInterface;`
3. Something else...

It'd be great to hear opinions on this.

Reply:
I think a set of booleans would be more explicit here:

```typescript
interface Alarms {
  5xxAlarm: boolean;
  unhealthyHostAlarm: boolean;
  anotherAlarm: boolean;
}
```

Downside is its another thing to maintain/keep up to date - maybe there's some funky generic TS we can write...